hnswlib/space_ip.h

```diff
@@ -245,6 +251,41 @@ namespace hnswlib {
 #endif
 
 #if defined(USE_SSE) || defined(USE_AVX) || defined(USE_AVX512)
+    static float
+    InnerProductSIMD4Ext(const void *pVect1v, const void *pVect2v, const void *qty_ptr) {
+        DISTFUNC<float> simdfunc_;
+    #if defined(USE_AVX)
+        if (AVXCapable())
+            simdfunc_ = InnerProductSIMD4ExtAVX;
+        else
+            simdfunc_ = InnerProductSIMD4ExtSSE;
+    #else
+        simdfunc_ = InnerProductSIMD4ExtSSE;
+    #endif
+        return simdfunc_(pVect1v, pVect2v, qty_ptr);
+    }
+
+    static float
+    InnerProductSIMD16Ext(const void *pVect1v, const void *pVect2v, const void *qty_ptr) {
+        DISTFUNC<float> simdfunc_;
+    #if defined(USE_AVX512)
+        if (AVX512Capable())
```

Comment:
It seems like it does a test for the AVX every distance calculation, which I guess would damage the performance.

Reply:
I'll take another look to see if I can only perform the check only once in the constructor. The difficulty in doing that is that the *ExtResiduals functions use the *Ext functions and are outside the class.

Reply:
Maybe this can be solved by adding a template parameter with avx support inside so it (so compiler would optimize the unused branches away)?

Reply:
I got a VM for windows and fixed windows build now.

Reply:
Thank you! I'll check it. Hopefully, this weekend.